keps/sig-api-machinery/1904-efficient-watch-resumption/README.md

```diff
@@ -267,48 +267,59 @@ delievered to a subset of kube-apiservers, this is not a problem, because:
   should still be stored there (unless there is heavy churn of those objects
   which is the case that doesn't suffer from this problem)
 
-
 The POC PR can be found in: https://github.com/kubernetes/kubernetes/pull/92472
 
 ### Risks and Mitigations
 
-<!--
-What are the risks of this proposal, and how do we mitigate? Think broadly.
-For example, consider both security and how this will impact the larger
-Kubernetes ecosystem.
-
-How will security be reviewed, and by whom?
-
-How will UX be reviewed, and by whom?
-
-Consider including folks who also work outside the SIG or subproject.
--->
+The biggest risk are bugs in the implementation. To mitigate this, the
+implementation will be hidden behind `EfficientWatchResumption` feature
+gate and necessary tests will be added and/or extended (details below).
 
 ## Design Details
 
 ### Test Plan
 
-TODO: Fill in before making `Implementable`.
+- [Alpha] unit tests for logic enhancing resource version tracking in reflector
+- [Alpha] unit tests for newly added watch cache logic
+- [Beta] Integration/e2e test for sending bookmark on kube-apiserver shutdown
+- [Beta] Integration/e2e test for proving that resource version that
```

Comment:
Integration sounds fine.

I don't think you need to e2e test this feature, I think integration tests will be easier to make exhaustive.

If you want an e2e test, I'd actually consider something that analyzes e.g. the audit log to count relists per watcher.

Reply:
Also I kind of expect these integration tests for alpha?

Reply:
Yeah - I agree that integration seems good enough.

Also switched to be required in alpha.